tests/integration/test_kfp_functional_v2.py

```diff
@@ -0,0 +1,229 @@
+        status="active",
+        raise_on_blocked=False,  # These apps block while waiting for each other to deploy/relate
+        raise_on_error=True,
+        timeout=3600,
+        idle_period=30,
+    )
+
+
+# ---- KFP API Server focused test cases
+async def test_upload_pipeline(kfp_client):
+    """Upload a pipeline from a YAML file and assert its presence."""
+    # Upload a pipeline and get the server response
+    pipeline_name = f"test-pipeline-sdk-{KFP_SDK_VERSION}"
+    pipeline_upload_response = kfp_client.pipeline_uploads.upload_pipeline(
+        uploadfile=SAMPLE_PIPELINE[KFP_SDK_VERSION],
+        name=pipeline_name,
+    )
+    # Upload a pipeline and get its ID
+    uploaded_pipeline_id = pipeline_upload_response.pipeline_id
+
+    # Get pipeline id by name, default='sample-pipeline'
+    server_pipeline_id = kfp_client.get_pipeline_id(name=pipeline_name)
+    assert uploaded_pipeline_id == server_pipeline_id
+
+
+async def test_create_and_monitor_run(kfp_client, create_and_clean_experiment_v2):
+    """Create a run and monitor it to completion."""
+    # Create a run, save response in variable for easy manipulation
+    # Create an experiment for this run
+    experiment_response = create_and_clean_experiment_v2
+
+    # Create a run from a pipeline file (SAMPLE_PIPELINE) and an experiment (create_experiment).
+    # This call uses the 'default' kubeflow service account to be able to edit Workflows
+    create_run_response = kfp_client.create_run_from_pipeline_package(
+        pipeline_file=SAMPLE_PIPELINE[KFP_SDK_VERSION],
+        arguments={},
+        run_name=f"test-run-sdk-{KFP_SDK_VERSION}",
+        experiment_name=experiment_response.display_name,
+        namespace=KUBEFLOW_PROFILE_NAMESPACE,
+    )
+
+    # Monitor the run to completion, the pipeline should not be executed in
+    # more than 300 seconds as it is a very simple operation
+    monitor_response = kfp_client.wait_for_run_completion(create_run_response.run_id, timeout=600)
+
+    assert monitor_response.state == "SUCCEEDED"
+
+
+# ---- ScheduledWorfklows and Argo focused test case
+async def test_create_and_monitor_recurring_run(
+    kfp_client, upload_and_clean_pipeline_v2, create_and_clean_experiment_v2
+):
+    """Create a recurring run and monitor it to completion."""
+
+    # Upload a pipeline from file
+    pipeline_response, pipeline_version_id = upload_and_clean_pipeline_v2
+
+    # Create an experiment for this run
+    experiment_response = create_and_clean_experiment_v2
+
+    # Create a recurring run from a pipeline (upload_pipeline_from_file) and an experiment (create_experiment)
+    # This call uses the 'default' kubeflow service account to be able to edit ScheduledWorkflows
+    # This ScheduledWorkflow (Recurring Run) will run once every two seconds
+    create_recurring_run_response = kfp_client.create_recurring_run(
+        experiment_id=experiment_response.experiment_id,
+        job_name="recurring-job-1",
```

Comment:
nit but should we name this `recurring-job-KFP_SDK_VERSION` since we 've also did
```
    pipeline_name = f"test-pipeline-sdk-{KFP_SDK_VERSION}"
...
        run_name=f"test-run-sdk-{KFP_SDK_VERSION}",
```

If yes, we could do that for v1 too.

Reply:
Sure, we could do that.